incubator/mongodb-replicaset/init/files/on-start.sh

```diff
@@ -14,48 +14,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-function join {
-    local IFS="$1"; shift; echo "$*";
+replica_set=$REPLICA_SET
+script_name=${0##*/}
+
+if [[ "$AUTH" == "true" ]]; then
+    admin_user="$ADMIN_USER"
+    admin_password="$ADMIN_PASSWORD"
+    admin_auth=(-u "$admin_user" -p "$admin_password")
+fi
+
+function log() {
+    local msg="$1"
+    local timestamp=$(date --iso-8601=ns)
+    echo "[$timestamp] [$script_name] $msg" >> /work-dir/log.txt
+}
+
+function shutdown_mongo() {
+    log "Shutting down MongoDB..."
+    mongo admin "${admin_auth[@]}" --eval 'db.shutdownServer({force: true})'
```

Comment:
Maybe we should allow specifying a timeout here for any secondaries to catch up? I imagine that this could lead to data loss in some situations.